### PR TITLE
Add collection list query

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this because it adds `SaleorAPI.collections.getList()` method and `useCollectionList` hook.

### CODE REVIEW

File: src/api/collections/CollectionList.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { ApolloQueryResult } from "apollo-client";
+import {
+  CollectionList as CollectionListQuery,
+  CollectionListVariables,
+} from "../../queries/gqlTypes/CollectionList";
+import { BaseCollection } from "../../fragments/gqlTypes/BaseCollection";
+import { PageInfo } from "../../fragments/gqlTypes/PageInfo";
+import { List } from "../types";
+
+export const COLLECTIONS_PER_API_CALL = 20;
```
Comment (orzechdev) [inline src/api/collections/CollectionList.ts:10]: Such variables would be passed in SDK config, optionally obtained from default config.

File: src/queries/index.ts [hunk 2]
```diff
@@ -60,6 +65,14 @@ export const QUERIES = {
       query: Category.categoryQuery,
       ...options,
     }),
+  CollectionList: (
```
Comment (orzechdev) [inline src/queries/index.ts:68]: Do we use it at all? Why is it defined here?

File: src/api/APIProxy.ts [hunk 3]
```diff
@@ -56,6 +57,15 @@ const handleDataErrors = <T extends QueryShape, TData>(
 };
 
 class APIProxy {
+  collections: CollectionsAPI;
```
[the remark this answers is not in the document]
Comment (dominik-zeglen) [reply src/api/APIProxy.ts:60]: Yeah prolly best place

**Comment by patrys** [comment]
Task linked: [CU-6a5vy3 Get collection list](https://app.clickup.com/t/6a5vy3)

**Pull request merged**